closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -261,13 +267,44 @@ static void printErrors(void)
     fflush(stderr);
 }
 
+/*
+ * We use a 8 digit map (ABBCCDDD) to represent the version of openssl.
+ * A is the major version,
+ * BB is the minor version,
+ * CC is the revision,
+ * DDD for the letter that could be present in any version.
+ * For example, if an openssl version is in this scheme 1.2.3.d
+ * where major is 1, minor is 2, reversion is 3 and last char is d -> 4.
+ * So the result would be 0x10203004.
+ */
+static jlong extractVersionToJlong(const char *astring)
+{
+    long major = 0;
+    long minor = 0;
+    long fix = 0;
+    long patch = 0;
+    char last_char = 0;
+    if (sscanf(astring, "OpenSSL %ld.%ld.%ld%c", &major, &minor, &fix, &last_char) < 3) {
+        return -1;
+    }
+    last_char = tolower(last_char);
+    if (('a' <= last_char) && (last_char <= 'i')) {
+        patch = last_char - 'a' + 1;
+    } else if (('j' <= last_char) && (last_char <= 'r')) {
+        patch = last_char - 'j' + 10;
+    } else if (('s' <= last_char) && (last_char <= 'z')) {
+        patch = last_char - 's' + 19;
+    }
```

Comment:
I thought about this some more; on z/OS this code will be compiled with `convlit(ISO8859-1)` so the literals will be ASCII (and `sscanf()` will do the appropriate conversions), so we can use the simpler version we had earlier:
```c
    if (isalpha(patch_char)) {
        patch = tolower(patch_char) - 'a' + 1;
    }
````

Reply:
Updated.